fairlearn/adversarial/_preprocessor.py

```diff
@@ -103,11 +103,17 @@ def _check(self, X, dtype=None, init=False):
         )
         if init:
             self.input_dim_ = X.ndim
+            if self.input_dim_ > 2:
+                raise ValueError(
```

Comment:
Sorry for the delay in reviewing, but is it possible to trigger this error in a test? Same for the `SystemError` below?